analytic_engine/src/sst/parquet/builder.rs

```diff
@@ -99,16 +94,13 @@ impl std::io::Seek for EncodingBuffer {
     }
 }
 
-impl TryClone for EncodingBuffer {
-    fn try_clone(&self) -> std::io::Result<Self> {
-        Ok(self.clone())
-    }
-}
-
 impl EncodingBuffer {
-    fn read(&self, read_buf: &mut [u8]) -> usize {
-        let mut inner = self.inner.lock().unwrap();
-        inner.read(read_buf)
+    fn into_bytes(self) -> Vec<u8> {
```

Comment:
The `EncodingBuffer` can be simplified (at least no need to wrap vector wth `Arc<Mutex>`) after the `RecordBytesReader` is refactored.